internal/dinosaur/pkg/services/quota/ams_quota_service_test.go

```diff
@@ -925,9 +929,245 @@ func Test_amsQuotaService_CheckIfQuotaIsDefinedForInstanceType(t *testing.T) {
+				return nil, fmt.Errorf("failed to get quota cost")
+			},
+			central:    standardCentral,
+			want:       false,
+			wantErr:    true,
+			wantErrMsg: "RHACS-MGMT-9: failed to get assigned quota of type \"standard\" for organization with external id \"\" and id \"fake-org-id-\"\n caused by: failed to get quota cost",
+		},
+		{
+			name: "returns an error when it finds only unsupported billing models",
+			getQuotaFunc: func(organizationID, resourceName, product string) ([]*v1.QuotaCost, error) {
+				rrbq := v1.NewRelatedResource().BillingModel("unsupported").Product(product).ResourceName(resourceName).Cost(1)
+				qcb, err := v1.NewQuotaCost().Allowed(allowed).Consumed(not_consumed).OrganizationID(organizationID).RelatedResources(rrbq).Build()
+				require.NoError(t, err)
+				return []*v1.QuotaCost{qcb}, nil
+			},
+			central:    standardCentral,
+			want:       false,
+			wantErr:    true,
+			wantErrMsg: "RHACS-MGMT-9: found only unsupported billing models [\"unsupported\"] for product \"RHACS\"",
+		},
+	}
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			g := gomega.NewWithT(t)
+
+			var amsClient ocm.AMSClient = &ocm.ClientMock{
+				GetOrganisationFromExternalIDFunc: makeOrganizationFromExtID,
+				GetQuotaCostsForProductFunc:       tt.getQuotaFunc,
+			}
+
+			quotaServiceFactory := NewDefaultQuotaServiceFactory(amsClient, nil, nil)
+			quotaService, _ := quotaServiceFactory.GetQuotaService(api.AMSQuotaType)
+
+			got, err := quotaService.HasQuotaAllowance(tt.central, types.DinosaurInstanceType(tt.central.InstanceType))
+			g.Expect(err != nil).To(gomega.Equal(tt.wantErr))
+			if tt.wantErr {
+				g.Expect(err.Error()).To(gomega.Equal(tt.wantErrMsg), err.Error())
+			}
+			g.Expect(got).To(gomega.Equal(tt.want))
+		})
+	}
+}
+
+func makeStandardTestQuotaCost(resourceName string, organizationID string, allowed int, consumed int, t *testing.T) *v1.QuotaCost {
+	rrbq := v1.NewRelatedResource().BillingModel(string(v1.BillingModelStandard)).Product(string(ocm.RHACSProduct)).ResourceName(resourceName).Cost(1)
+	qcb, err := v1.NewQuotaCost().Allowed(allowed).Consumed(consumed).OrganizationID(organizationID).RelatedResources(rrbq).Build()
+	require.NoError(t, err)
+	return qcb
+}
+
+func makeCloudTestQuotaCost(resourceName string, organizationID string, allowed int, consumed int, t *testing.T) *v1.QuotaCost {
+	cloudAccount := v1.NewCloudAccount().CloudAccountID("cloudAccountID").CloudProviderID(awsCloudProvider)
+	rrbq := v1.NewRelatedResource().BillingModel(string(v1.BillingModelMarketplaceAWS)).Product(string(ocm.RHACSProduct)).ResourceName(resourceName).Cost(1)
+	qcb, err := v1.NewQuotaCost().Allowed(allowed).Consumed(consumed).OrganizationID(organizationID).RelatedResources(rrbq).CloudAccounts(cloudAccount).Build()
+	require.NoError(t, err)
+	return qcb
+}
+
+func makeOrganizationFromExtID(externalId string) (*v1.Organization, error) {
```

Comment:
```suggestion
func makeOrganizationFromExternalID(externalId string) (*v1.Organization, error) {
```

We can allow ourselves 5 more letters 😉